integration-tests/elasticsearch-rest-client/pom.xml

```diff
@@ -91,35 +97,94 @@
+        <elasticsearch.protocol>http</elasticsearch.protocol>
+      </properties>
+      <build>
+        <plugins>
           <plugin>
-            <groupId>com.github.alexcojocaru</groupId>
-            <artifactId>elasticsearch-maven-plugin</artifactId>
+            <groupId>io.fabric8</groupId>
+            <artifactId>docker-maven-plugin</artifactId>
             <configuration>
-              <timeout>90</timeout>
-              <pathConf>${project.build.directory}/test-classes/elasticsearch-maven-plugin/configuration/</pathConf>
-              <autoCreateIndex>true</autoCreateIndex>
+              <images>
+                <image>
+                  <name>${elasticsearch.image}</name>
+                  <alias>elasticsearch</alias>
+                  <run>
+                    <env>
+                      <discovery.type>single-node</discovery.type>
+                    </env>
+                    <ports>
+                      <port>9200:9200</port>
+                    </ports>
+                    <log>
+                      <prefix>Elasticsearch: </prefix>
+                      <date>default</date>
+                      <color>cyan</color>
+                    </log>
+                    <wait>
+                      <http>
+                        <url>http://localhost:9200</url>
+                        <method>GET</method>
+                        <status>200</status>
+                      </http>
+                      <time>20000</time>
+                    </wait>
+                  </run>
+                </image>
+              </images>
+              <allContainers>true</allContainers>
             </configuration>
-            <!-- Different executions from 2.x: the "start" goal has been renamed in version 5  -->
             <executions>
               <execution>
-                <id>start-elasticsearch</id>
+                <id>docker-start</id>
                 <phase>process-test-classes</phase>
                 <goals>
-                  <goal>runforked</goal>
+                  <goal>stop</goal>
+                  <goal>start</goal>
                 </goals>
               </execution>
               <execution>
-                <id>stop-elasticsearch</id>
+                <id>docker-stop</id>
                 <phase>post-integration-test</phase>
                 <goals>
                   <goal>stop</goal>
                 </goals>
               </execution>
             </executions>
           </plugin>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>exec-maven-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>docker-prune</id>
+                <phase>generate-resources</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>${basedir}/../../.github/docker-prune.sh</executable>
```

Comment:
Interesting, I just learned about this script.

Reply:
I copy-pasted that from another pom file. Not sure it's useful, but well... that's how it works everywhere else :shrug: